streaming/src/main/scala/org/apache/spark/streaming/api/java/JavaPairDStream.scala

```diff
@@ -789,7 +790,7 @@ class JavaPairDStream[K, V](val dstream: DStream[(K, V)])(
       keyClass: Class[_],
       valueClass: Class[_],
       outputFormatClass: Class[_ <: NewOutputFormat[_, _]],
-      conf: Configuration = new Configuration) {
```

Comment:
The scope of this PR is pretty wide in terms of the number of classes it touches, causing issues as different places needs to be handled differently. If you considered moving this sort of changes (`new Configuration` to `sparkContext.hadoopConfiguration`) into a different PR that might be easier to get in.